src/ir/operation.cc

```diff
@@ -23,22 +23,61 @@
 
 namespace raksha::ir {
 
+// Note: I tried to make this only defined in terms of the type parameter T, but
+// the compiler was unhappy with me writing std::function<std::string(const T&)>
+// and I could not figure out how to placate it.
+template<class T, class U>
+static std::string PrintNamedMapInNameOrder(
+    const absl::flat_hash_map<std::string, T> &map_to_print,
+    U value_printing_fn) {
+  std::vector<absl::string_view> names;
+  names.reserve(map_to_print.size());
+  for (auto &key_value_pair : map_to_print) {
+    names.push_back(key_value_pair.first);
+  }
+  std::sort(names.begin(), names.end());
+  return absl::StrJoin(
+      names, ", ",
+      [&](std::string* out, const absl::string_view name) {
+        auto find_res = map_to_print.find(name);
+        CHECK(find_res != map_to_print.end())
+          << "Name " << name << " not found in map_to_print";
+        absl::StrAppend(out, name, ": ", value_printing_fn(find_res->second));
+      });
+}
+
+std::string Operation::NoSsaToString() const {
```

Comment:
I started refactoring the `ToString` method as a visitor here: https://github.com/google-research/raksha/blob/arcs_to_ir%40bgogul/src/ir/ir_printer.h

Reply:
I have re-written the test in a more visitor style, but I didn't use the visitor you wrote. I realized, while trying to adapt your visitor to my problem, that it isn't the right tool for the job because I wanted to visit values in dataflow order. In general, that starts to look a lot like a dataflow analysis. For this problem specifically, however, the values should just linearly nest. So I wrote a special purpose visitor and used some downcast methods.